backend/src/friends/friends.service.ts

```diff
@@ -0,0 +1,92 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { User } from '../users/entities/user.entity';
+import { UsersService } from '../users/users.service';
+import { FRIEND_STATUS, Friend } from './entities/friend.entity';
+
+@Injectable()
+export class FriendsService {
+  constructor(
+    @InjectRepository(Friend)
+    private readonly friendRepository: Repository<Friend>,
+    private readonly usersService: UsersService,
+  ) {}
+
+  // TODO: Split into received and sent requests
+  async listRequests(userId: number): Promise<Friend[]> {
+    return await this.friendRepository.find({
+      where: [
+        { sender: { id: userId }, status: FRIEND_STATUS.PENDING },
+        { recipient: { id: userId }, status: FRIEND_STATUS.PENDING },
+      ],
+      relations: ['sender', 'recipient'],
+    });
+  }
+
+  async listFriends(userId: number): Promise<User[]> {
+    const acceptedFriendRequests = await this.friendRepository.find({
+      where: [
+        { sender: { id: userId }, status: FRIEND_STATUS.ACCEPTED },
+        { recipient: { id: userId }, status: FRIEND_STATUS.ACCEPTED },
+      ],
+      relations: ['sender', 'recipient'],
+    });
+
+    return acceptedFriendRequests.map((request) =>
+      request.sender.id !== userId ? request.sender : request.recipient,
+    );
+  }
+
+  async store(userId: number, friendId: number): Promise<Friend> {
+    const friend = new Friend();
+
+    const user = await this.usersService.findOne(userId);
+    const futureFriend = await this.usersService.findOne(friendId);
+
+    friend.sender = user;
+    friend.recipient = futureFriend;
+
+    await friend.save();
+
+    return friend;
+  }
+
+  async accept(id: number): Promise<Friend> {
+    const friend_request = await this.friendRepository.findOneBy({ id });
+
+    friend_request.status = FRIEND_STATUS.ACCEPTED;
+    friend_request.accepted_date = new Date();
+
+    await friend_request.save();
+
+    return friend_request;
+  }
+
+  async reject(id: number): Promise<void> {
+    const friendRequest = await this.friendRepository.findOneBy({ id });
+
+    if (!friendRequest)
+      throw new NotFoundException(`Friend request Id $(id) not found`);
+
+    await this.friendRepository.remove(friendRequest);
```

Comment:
should we remove the friend_request from db or change its status to REJECTED?

Reply:
That's a good one, the way it's structured now, it'd be harder to sort stuff out if we let it as REJECTED. If some denies a user, and the "rejectee" tries to send another friend request, it'd be denied to because the request would already be sent.

Surely we could make it work tweaking the service a lil bit, but deleting the entry solves us some problem for now.